### PR TITLE
Add djgpp support

### PR DESCRIPTION
**Pull request by phcoder**
This is the main part of DJGPP support as libretro core. We also need some changes from duktape (https://github.com/svaarala/duktape/issues/2472) and wren (https://github.com/wren-lang/wren/pull/1090)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -623,9 +623,23 @@ if(BUILD_LIBRETRO)
             set(LIBRETRO_EXTENSION "a")
         endif()
         set_target_properties(tic80_libretro PROPERTIES SUFFIX "_partial.a")
-        add_custom_command(TARGET tic80_libretro
+        include(CheckCSourceCompiles)
+        check_c_source_compiles(
+"#ifndef __MSDOS__
+#error \"Not DOS\"
```
Comment (joshgoebel) [inline CMakeLists.txt:629]: Wait, is DJGPP a DOS compiler?  Are you building TIC-80 on DOS?

File: CMakeLists.txt [hunk 2]
```diff
@@ -623,9 +623,23 @@ if(BUILD_LIBRETRO)
...
+#endif"
+IS_DOS
+)
+
+        if(IS_DOS)
+          add_custom_command(TARGET tic80_libretro
+               POST_BUILD
+                   COMMAND ${CMAKE_SOURCE_DIR}/build/libretro/merge_static.sh $(AR) ${CMAKE_BINARY_DIR}/lib/tic80_libretro${LIBRETRO_SUFFIX}.${LIBRETRO_EXTENSION} ${CMAKE_BINARY_DIR}/lib/tic80_libretro_partial.a ${CMAKE_BINARY_DIR}/lib/libtic80core.a ${CMAKE_BINARY_DIR}/lib/liblua.a ${CMAKE_BINARY_DIR}/lib/libblipbuf.a ${CMAKE_BINARY_DIR}/lib/libduktape.a ${CMAKE_BINARY_DIR}/lib/libwren.a ${CMAKE_BINARY_DIR}/lib/libwasm.a ${CMAKE_BINARY_DIR}/lib/libsquirrel.a ${CMAKE_BINARY_DIR}/lib/libgiflib.a ${CMAKE_BINARY_DIR}/lib/liblpeg.a ${CMAKE_BINARY_DIR}/lib/libzlib.a ${MRUBY_LIB})
```
Comment (joshgoebel) [inline CMakeLists.txt:637]: Why do we only need zlib on DOS?

File: CMakeLists.txt (hunk 1 above)
Comment (phcoder) [reply CMakeLists.txt:629]: For DOS. Cross-compiling from Debian

File: CMakeLists.txt (hunk 2 above)
Comment (phcoder) [reply CMakeLists.txt:637]: * dynamically linked libretro platforms don't end up in this code path
* Other statically linked ones pull in system zlib via -lz
* DOS is special that it doesn't have system or SDK-provided zlib

Comment (phcoder) [reply CMakeLists.txt:637]: Addendum: it will also be needed for NGC and Wii but they need endianness patch in first

**Pull request merged**